### PR TITLE
Fixed not being able to delete avatar entities when switching domains

### PR DESCRIPTION
**Pull request by druiz17**
When you have an avatar entity. Using Entities.deleteEntity() on a domain switch does not work because the interface has not yet set the entity ownership to you. So deleteEntity function has the entity ownership to {00000000-0000-0000-0000-000000000000}, thus thinking you do not own the entity.

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- run the script below (the script should add a large white box at your avatar position)
- switch domains
- the box should be deleted
- reset the scripts and perform the test again. (do this for a least 5 times)

**Comment by CainFoool** [comment]
Testing once built.

**Comment by CainFoool** [comment]
@druiz17: It looks like you didn't attach the script to use.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1723/HighFidelity-Beta-PR10981-21d760a533027610c8717c626a3052c7177b770d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1723/HighFidelity-Beta-PR10981-21d760a533027610c8717c626a3052c7177b770d.dmg

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
run the script below (the script should add a large white box at your avatar position) | Pass
switch domains | Pass
the box should be deleted | Pass
reset the scripts and perform the test again. (do this for a least 5 times) | Pass

Tried this 10 times and the box deleted itself each time with no issue upon switching domains.

Let me know if there's anything else to do here.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -497,8 +498,13 @@ void EntityScriptingInterface::deleteEntity(QUuid id) {
 
                 auto nodeList = DependencyManager::get<NodeList>();
                 const QUuid myNodeID = nodeList->getSessionUUID();
+                auto avatarHashMap = DependencyManager::get<AvatarHashMap>();
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: these should be inside the `if (if (entity->getClientOnly() && entity->getOwningAvatarID() != myNodeID) {` clause below, don't do this work for non-client entities

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 2]
```diff
@@ -497,8 +498,13 @@ void EntityScriptingInterface::deleteEntity(QUuid id) {
...
+                AvatarSharedPointer myAvatar = avatarHashMap->getAvatarBySessionID(myNodeID);
                 if (entity->getClientOnly() && entity->getOwningAvatarID() != myNodeID) {
                     // don't delete other avatar's avatarEntities
+                    // If you actually own the entity but the onwership is not set because of a domain switch
+                    // the line below make sure the entity is deleted.
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: this looks like a typo... 

> // the line below make*s* sure the entity is deleted.

**Comment by AndrewMeadows** [comment]
Jenkins, rebuild this please.

**Comment by druiz17** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1751/HighFidelity-Beta-PR10981-7fe47e4e2f7258a35536a51a1b8e53b1e1eb355e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1751/HighFidelity-Beta-PR10981-7fe47e4e2f7258a35536a51a1b8e53b1e1eb355e.dmg

**Pull request merged**